### PR TITLE
pythonPackages.simplefix: init at 1.0.12

### PR DESCRIPTION
**Pull request by catern**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
None of the tests get executed by this package as it is. To get them to work you've got to change the src to the github archive (which includes the tests):

```
  src = fetchFromGitHub {
    repo = "simplefix";
    owner = "da4089";
    rev = "v${version}";
    sha256 = "0pnyqxpki1ija0kha7axi6irgiifcz4w77libagkv46b1z11cc4r";
  };
```

and add

```
  checkPhase = ''
    cd test
    ${python.interpreter} -m unittest all
  '';
```

**Comment by catern** [comment]
Great point, updated as you suggest to discover the tests; they all pass.

File: pkgs/development/python-modules/simplefix/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    rev = "v${version}";
+    sha256 = "0pnyqxpki1ija0kha7axi6irgiifcz4w77libagkv46b1z11cc4r";
+  };
+
+  checkPhase = ''
+    cd test
+    ${python.interpreter} -m unittest all
+  '';
+
+  meta = {
```
Comment (jonringer) [inline pkgs/development/python-modules/simplefix/default.nix]: personal preference, don't feel strongly about this.
```suggestion
  meta = with lib; {
```

File: pkgs/development/python-modules/simplefix/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    description = "Simple FIX Protocol implementation for Python";
+    homepage = https://github.com/da4089/simplefix;
+    license = lib.licenses.mit;
```
Comment (jonringer) [inline pkgs/development/python-modules/simplefix/default.nix]:
```suggestion
    license = licenses.mit;
```

File: pkgs/development/python-modules/simplefix/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    maintainers = with lib.maintainers; [ catern ];
```
Comment (jonringer) [inline pkgs/development/python-modules/simplefix/default.nix]:
```suggestion
    maintainers = with maintainers; [ catern ];
```

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
leaf package

I wasn't able to get the tests to run myself. But you followed the directions of the repo, so not sure what to do :(.

**Comment by worldofpeace** [comment]
Feel free to ping me for merge when @jonringer changes are applied.

**Comment by catern** [comment]
@worldofpeace Applied those changes and the checks have passed

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build python27Packages.simplefix python35Packages.simplefix python36Packages.simplefix python37Packages.simplefix

**Comment by worldofpeace** [comment]
Thanks @catern :sparkles:

**Pull request merged**